src/wp-admin/includes/image.php

```diff
@@ -914,9 +913,52 @@ function wp_generate_attachment_metadata( $attachment_id, $file ) {
 					// Save the meta data before any image post-processing errors could happen.
 					wp_update_attachment_metadata( $attachment_id, $metadata );
 
-					// Create sub-sizes saving the image meta after each.
-					$metadata = _wp_make_subsizes( $merged_sizes, $image_file, $metadata, $attachment_id, '' );
+					$primary_mime_type = isset( $metadata['sizes']['full']['mime-type'] ) ? $metadata['sizes']['full']['mime-type'] : 'image/jpeg';
+
+					// Get mime types to generate.
+					$transforms      = wp_upload_image_mime_transforms( $attachment_id );
+					$mime_transforms = array();
+
+					if ( ! empty( $transforms[ $primary_mime_type ] ) ) {
+						$mime_transforms = $transforms[ $primary_mime_type ];
+					}
+
+					// Generate image sub-sizes for each mime type.
+					if ( ! empty( $mime_transforms ) ) {
+						foreach ( $mime_transforms as $mime_type ) {
+							if ( ! $editor::supports_mime_type( $mime_type ) ) {
+								continue;
+							}
+
+							if ( $primary_mime_type === $mime_type ) {
+								// Add primary mime type sources.
+								$metadata['sizes']['full']['sources'][ $primary_mime_type ] = _wp_get_sources_from_meta( $uploaded );
+							} else {
+								$extension = wp_get_default_extension_for_mime_type( $mime_type );
+								if ( false === $extension ) {
+									continue;
+								}
+
+								$editor->set_output_mime_type( $mime_type );
+								$result = $editor->save( $editor->generate_filename( '', pathinfo( $image_file, PATHINFO_FILENAME ), $extension ) );
```

Comment:
do you need to generate the name here? `$editor->save()` should use the correct naming once you set the output mime type

Reply:
when we not generate the name by just using `$editor->save()`, it adds suffix of image resolution to the name.

Reply:
👍🏼 hmm, that might be ok to have the dimensions suffix? I'll review again